Allow emoji aliases to be created from emojipacks

IgnoredAmbience opened the pull request:
Reopening #52 with refreshed patch. Might conflict with #105, but I'm happy to rebase against it again.

Aliases can be defined using a list of names under the aliases key of an
emoji definition. See the updated README.md for more details.

New errors added to image-checker.py:
* `Error: Emoji named ___ already defined elsewhere.`
  For when the emoji being tested has been defined as an emoji or alias
  already.
* `Error: Alias named ___ already defined elsewhere.`
  For when the alias being tested has been defined as an emoji or alias
  already.
* `Warning: ___ should be an alias for ___.`
  For when two emoji definitions share the same URL.

Existing emojipacks have been aliased and de-duplicated as appropriate.

zachfeldman commented:
![image](https://cloud.githubusercontent.com/assets/1437644/25661874/59b12e3c-2fe0-11e7-9fc3-519e1bab32f1.png)

This contains new code that you just squashed right? Just making sure! @edgemaster

IgnoredAmbience commented:
Ah yes, that's just reusing the previous timestamp from #52. I should probably bump it to today.

IgnoredAmbience committed:
Allow emoji aliases to be created from emojipacks

Aliases can be defined using a list of names under the aliases key of an
emoji definition. See the updated README.md for more details.

New errors added to image-checker.py:
* `Error: Emoji named ___ already defined elsewhere.`
  For when the emoji being tested has been defined as an emoji or alias
  already.
* `Error: Alias named ___ already defined elsewhere.`
  For when the alias being tested has been defined as an emoji or alias
  already.
* `Warning: ___ should be an alias for ___.`
  For when two emoji definitions share the same URL.

Existing emojipacks have been aliased and de-duplicated as appropriate.

IgnoredAmbience commented:
Rebased and updated timestamp.

zachfeldman commented:
Cool looks good.
